integrations/tensorflow/e2e/math/math_test.py

```diff
@@ -0,0 +1,570 @@
+    "sinh":
+        DtypeConfig(signature=UNARY, dtypes=FLOATING_NUMBER_TYPES),
+    "sobol_sample":
+        ArgsConfig(args=[4, 3]),
+    "softmax":
+        Config(signature=UNARY_FLOAT),
+    "softplus":
+        Config(signature=UNARY_FLOAT),
+    "softsign":
+        Config(signature=UNARY_FLOAT),
+    "sqrt":
+        DtypeConfig(signature=UNARY, dtypes=FLOATING_NUMBER_TYPES),
+    "square":
+        DtypeConfig(signature=UNARY, dtypes=ALL_NUMBER_TYPES),
+    "squared_difference":
+        DtypeConfig(signature=BINARY, dtypes=ALL_NUMBER_TYPES),
+    "subtract":
+        DtypeConfig(signature=BINARY, dtypes=ALL_NUMBER_TYPES),
+    "tan":
+        DtypeConfig(signature=UNARY, dtypes=FLOATING_NUMBER_TYPES),
+    "tanh":
+        DtypeConfig(signature=UNARY, dtypes=FLOATING_NUMBER_TYPES),
+    "top_k":
+        Config(signature=UNARY_FLOAT, kwargs=dict(k=2)),
+    "truediv":
+        DtypeConfig(signature=BINARY, dtypes=FLOATING_NUMBER_TYPES),
+    "unsorted_segment_max":
+        UNSORTED_SEGMENT_CONFIG,
+    "unsorted_segment_mean":
+        UNSORTED_SEGMENT_CONFIG,
+    "unsorted_segment_min":
+        UNSORTED_SEGMENT_CONFIG,
+    "unsorted_segment_prod":
+        UNSORTED_SEGMENT_CONFIG,
+    "unsorted_segment_sqrt_n":
+        UNSORTED_SEGMENT_CONFIG,
+    "unsorted_segment_sum":
+        UNSORTED_SEGMENT_CONFIG,
+    "xdivy":
+        DtypeConfig(signature=BINARY, dtypes=FLOATING_NUMBER_TYPES),
+    "xlog1py":
+        DtypeConfig(signature=BINARY, dtypes=FLOATING_NUMBER_TYPES),
+    "xlogy":
+        DtypeConfig(signature=BINARY, dtypes=FLOATING_NUMBER_TYPES),
+    "zero_fraction":
+        DtypeConfig(signature=UNARY, dtypes=ALL_NUMBER_TYPES),
+    "zeta":
+        Config(
+            signature=BINARY_FLOAT,
+            # The function is poorly behaved near zero, so we test this range
+            # to avoid outputing all nans.
+            generator=lambda *args: tf_utils.uniform(*args, low=3.0, high=4.0)),
+}
+
+
+def normalize_functions_to_configs(
+    functions_to_configs: Dict[str, Union[Config, Dict[str, Config]]]
+) -> Dict[str, Dict[str, Config]]:
+  # Normalize FUNCTION_TO_CONFIGS as described above.
+  normalized = dict()
+  for function_name, configs in functions_to_configs.items():
+    normalized[function_name] = dict()
+    if isinstance(configs, Config):
+      normalized[function_name] = {function_name: configs}
+    elif isinstance(configs, dict):
+      # Prepend the function names to existing exported names to make the logs
+      # readable. Split into two expressions so pytype can understand it.
+      normalized[function_name] = {
+          f"{function_name}_{config_name}": config
+          for config_name, config in configs.items()
+      }
+    else:
+      raise TypeError(
+          f"Unexpected type for value of FUNCTION_TO_CONFIGS {type(configs)}")
+  return normalized
+
+
+FUNCTION_TO_CONFIGS = normalize_functions_to_configs(FUNCTION_TO_CONFIGS)
```

Comment:
If you defined `normalize_functions_to_configs` first you could just call it passing an anonymous dict rather than redefining FUNCTION_TO_CONFIGS. That feels a bit nicer to me because the type doesn't change, but maybe I'm just biased towards static typing.